cpp/src/sampling/neighbor_sampling_impl.hpp

```diff
@@ -48,47 +52,68 @@ std::tuple<rmm::device_uvector<vertex_t>,
+                     std::optional<edge_property_view_t<edge_t, weight_t const*>> edge_weight_view,
+                     std::optional<edge_property_view_t<edge_t, edge_t const*>> edge_id_view,
+                     std::optional<edge_property_view_t<edge_t, edge_type_t const*>> edge_type_view,
+                     std::optional<edge_property_view_t<edge_t, bias_t const*>> edge_bias_view,
+                     raft::device_span<vertex_t const> starting_vertices,
+                     std::optional<raft::device_span<label_t const>> starting_vertex_labels,
+                     std::optional<raft::device_span<int32_t const>> label_to_output_comm_rank,
+                     raft::host_span<int32_t const> fan_out,
+                     edge_type_t num_edge_types,
+                     bool return_hops,
+                     bool with_replacement,
+                     prior_sources_behavior_t prior_sources_behavior,
+                     bool dedupe_sources,
+                     bool do_expensive_check)
 {
-#ifdef NO_CUGRAPH_OPS  // FIXME: this is relevant only when edge_bias_view.has_value() is false,
-                       // this ifdef statement will be removed once we migrate relevant cugraph-ops
-                       // functions to cugraph
-  CUGRAPH_FAIL(
-    "neighbor_sample_impl not supported in this configuration, built with NO_CUGRAPH_OPS");
-#else
   static_assert(std::is_floating_point_v<bias_t>);
 
+  std::vector<cugraph::edge_property_t<graph_view_t<vertex_t, edge_t, store_transposed, multi_gpu>, bool>> edge_masks_vector{};
+  graph_view_t<vertex_t, edge_t, false, multi_gpu> modified_graph_view = graph_view;
+
+
   CUGRAPH_EXPECTS(fan_out.size() > 0, "Invalid input argument: number of levels must be non-zero.");
   CUGRAPH_EXPECTS(
     fan_out.size() <= static_cast<size_t>(std::numeric_limits<int32_t>::max()),
     "Invalid input argument: number of levels should not overflow int32_t");  // as we use int32_t
                                                                               // to store hops
+  edge_masks_vector.reserve(num_edge_types);
+  
+  for (int i = 0; i < num_edge_types; i++) {
+
+    cugraph::edge_property_t<graph_view_t<vertex_t, edge_t, store_transposed, multi_gpu>, bool> edge_mask(handle, graph_view);
+    
+    cugraph::fill_edge_property(handle, modified_graph_view, edge_mask.mutable_view(), bool{true});
+
+    modified_graph_view.attach_edge_mask(edge_mask.view());
+
+    cugraph::transform_e(
+      handle,
+      modified_graph_view,
+      cugraph::edge_src_dummy_property_t{}.view(),
+      cugraph::edge_dst_dummy_property_t{}.view(),
+      *edge_type_view,
+      [valid_edge_type = i] __device__(auto src, auto dst, thrust::nullopt_t, thrust::nullopt_t, auto edge_type) {
+        return edge_type == valid_edge_type;
+      },
+      edge_mask.mutable_view(),
+      false);
+    
+    edge_masks_vector.push_back(std::move(edge_mask));
+  }
 
   if constexpr (!multi_gpu) {
```

Comment:
These error checks (including the expensive ones) should be before we create the `edge_mask_vector`.